test/loader.js

```diff
@@ -0,0 +1,102 @@
+var fs = require('fs');
+var path = require('path');
+var loader = require('../index.js');
+
+var source = path.join(__dirname, './fixtures/Foo.elm');
+var expected = fs.readFileSync(path.join(__dirname, './expected/Foo.js'), 'utf-8');
+
+var toString = Object.prototype.toString;
+
+var isArray = function (obj) {
+  return toString.call(obj) === '[object Array]';
+};
+
+var mock = function (query, opts, callback) {
+  var emittedError;
+  var emittedWarning;
+
+  var result = {
+    loaders: [],
+    loaderIndex: 0,
+
+    resource: source,
+    resourcePath: source,
+
+    async: function () { return callback; },
+
+    emitError: function (err) { emittedError = err; },
+    emittedError: function () { return emittedError; },
+    emitWarning: function (warn) { emittedWarning = warn; },
+    emittedWarning: function () { return emittedWarning; },
+
+    addDependency: function () {},
+    cacheable: function () {},
+
+    options: {}
+  };
+
+  if (query) {
+    result.query = '?' + (isArray(query) ? query.join('&') : query);
+  }
+
+  if (opts) {
+    result.options.ulmus = opts;
+  }
+
+  return result;
+};
+
+module.exports.test = {
+  'sync mode': {
+    'throws': function (test) {
+      var context = mock();
+
+      try {
+        loader.call(context, source);
+      }
+      catch (err) {
+        test.done();
+      }
+    }
+  },
+
+  'async mode': {
+    'compiles the resource': function (test) {
+      var context;
+
+      var callback = function (err, result) {
+        test.equal(null, err);
+        test.equal(result.trim(), expected.trim());
```

Comment:
If you do this, then the error message that comes out is crazy scary and kills all other error messages. How about checking a hash of the file contents instead?